agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/feature/model/events/agentEvents.kt

```diff
@@ -10,16 +11,42 @@ import kotlinx.serialization.Serializable
  * This event provides details about the agent's strategy, making it useful for
  * monitoring, debugging, and tracking the lifecycle of AI agents within the system.
  *
+ * @property eventId A unique identifier for the event or a group of events;
+ * @property executionInfo Provides contextual information about the execution associated with this event.
  * @property agentId The unique identifier of the AI agent;
  * @property runId The unique identifier of the AI agen run;
  * @property timestamp The timestamp of the event, in milliseconds since the Unix epoch.
  */
 @Serializable
 public data class AgentStartingEvent(
+    override val eventId: String,
+    override val executionInfo: AgentExecutionInfo,
     val agentId: String,
     val runId: String,
     override val timestamp: Long = Clock.System.now().toEpochMilliseconds(),
-) : DefinedFeatureEvent()
+) : DefinedFeatureEvent() {
+
+    /**
+     * @deprecated. Creates an instance of [AgentStartingEvent].
+     * Note! Do not relay on [executionInfo] parameter in this constructor.
+     */
+    @Deprecated(
+        message = "Please use constructor with executionInfo parameter",
+        replaceWith = ReplaceWith("AgentStartingEvent(executionInfo, agentId, runId)")
+    )
```

Comment:
Don't you want to remove it completely? We still can afford to make some breaking changes and not to contain deprecated api

Reply:
Okay